Assets/MixedRealityToolkit/_Core/Utilities/ClipPlane.cs

```diff
@@ -168,7 +178,14 @@ private Material GetMaterial(Renderer renderer)
             }
             else
             {
-                return renderer.material;
+                Material material = renderer.material;
```

Comment:
Any time you call `renderer.material` you generate another material. You should just be able to loop the list and destroy them as is.

Reply:
Unity only generates a new material the first time you call `renderer.material` then it is smart enough to realize an instance has already been created and returns that for later calls.

Reply:
> then it is smart enough to realize an instance has already been created and returns that for later calls

You'd think that, but it's actually the worst memory leak in Unity.

https://docs.unity3d.com/ScriptReference/Renderer-material.html

>Note:
This function automatically instantiates the materials and makes them unique to this renderer. **It is your responsibility to destroy the materials when the game object is being destroyed**. Resources.UnloadUnusedAssets also destroys the materials but it is usually only called when loading a new level.

[Use `renderer.GetMaterials()` instead](Renderer.GetMaterials)

Reply:
From your first link:

>If the material is used by any other renderers, this will clone the shared material and start using it from now on.

Reply:
Indeed, the cloned instance still needs to be cleaned up.

Reply:
Is this not doing that?

https://github.com/Cameron-Micka/MixedRealityToolkit-Unity/blob/8499045c2f442a2d840c6aa5f04500e8a75615ec/Assets/MixedRealityToolkit/_Core/Utilities/ClipPlane.cs#L107

Reply:
I just tried this in an empty project:

    public Renderer testRenderer;

    private void Update()
    {
        if (Input.anyKeyDown)
        {
            Material material = testRenderer.material;
            material.color = Color.red;
        }
    }

And, my material count went up by one (Window->Profiler->Memory->Materials) the first time any key was hit, then maintained constant (material count and memory wise) with sequential key strokes. @StephenHodgson are you see a different behavior?

Reply:
It's better to be paranoid than leaking!

Reply:
> are you see a different behavior?

Nope I'm seeing the same.